solutions/day3.py

```diff
@@ -0,0 +1,61 @@
+
+def read_input(filename: str) -> list:
+    # Read file line by line, store in list
+    input_list = []
+
+    with open(filename, "r") as f:
+        for line in f:
+            input_list.append(line.strip())
+    
+    return input_list
+
+
+def check_mul(memory: str, index: int) -> int:
+    # Store characters after 'mul(', stop at either a ')' or
+    # if temp_string is too long to be valid
+
+    temp_string = ""
+
+    while (memory[index] != ")") and (len(temp_string) < 8):
+        temp_string += memory[index]
+        index += 1
+
+    # Try to extract 2 numbers from temp_string and multiply them, return 0 for exceptions
+    try:
+        nums = [int(i) for i in temp_string.split(",")]
+        return nums[0] * nums[1]
+    except ValueError:
+        return 0
+    except IndexError:
+        return 0
```

Comment:
I checked, and this also works on my input, but note that it would give an incorrect result for `mul(2,2,3)` which should be invalid, but here it will return `4`

Reply:
You could fix it by adding `if len(nums) != 2: return 0`